jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -284,6 +308,36 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    def cone_aperture(self):
+        if not self._app._jdaviz_helper._loaded_mask_cube:
+            snackbar_message = SnackbarMessage(
+                "Cannot create cone aperture without valid mask cube loaded.",
+                color="error",
+                sender=self)
+            self.hub.broadcast(snackbar_message)
+            return
+        # Retrieve flux cube and create an array to represent the cone mask
+        flux_cube = self._app._jdaviz_helper._loaded_flux_cube.get_object(cls=Spectrum1D,
+                                                                          statistic=None)
+        masks_boolean_values = np.zeros_like(flux_cube.flux.value)
+
+        # Center is reverse coordinates
+        center = (self.aperture.selected_spatial_region.center.y,
+                  self.aperture.selected_spatial_region.center.x)
+        # TODO: Replace with code for retrieving display_unit in cubeviz when it is available
+        display_unit = flux_cube.spectral_axis.unit
+
+        # Loop through cube and create cone aperture at each wavelength. Then convert that to a
+        # mask using the selected aperture method and add that to a mask cube.
+        for index, wavelength in enumerate(flux_cube.spectral_axis):
+            radius = ((wavelength.to(display_unit).value / self.reference_wavelength) *
```

Comment:
At this point, will `spectral_axis` always be in wavelength, or is there a chance for it to be frequency or wave number?

If it is allowed to be frequency or wave number, is this formula still correct, since both are inversed proportional to wavelength?